src/webui/api/torrentscontroller.cpp

```diff
@@ -823,23 +824,35 @@ void TorrentsController::editTrackerAction()
     if (!newTrackerUrl.isValid())
         throw APIError(APIErrorType::BadParams, u"New tracker URL is invalid"_s);
 
-    QVector<BitTorrent::TrackerEntry> trackers = torrent->trackers();
+    const QList<BitTorrent::TrackerEntryStatus> currentTrackers = torrent->trackers();
+    QList<BitTorrent::TrackerEntry> entries;
+    entries.reserve(currentTrackers.size());
+
     bool match = false;
-    for (BitTorrent::TrackerEntry &tracker : trackers)
+    for (const BitTorrent::TrackerEntryStatus &tracker : currentTrackers)
     {
         const QUrl trackerUrl {tracker.url};
+
         if (trackerUrl == newTrackerUrl)
             throw APIError(APIErrorType::Conflict, u"New tracker URL already exists"_s);
+
+        BitTorrent::TrackerEntry entry
+        {
+            .url = tracker.url,
+            .tier = tracker.tier
+        };
+
         if (trackerUrl == origTrackerUrl)
         {
             match = true;
-            tracker.url = newTrackerUrl.toString();
+            entry.url = newTrackerUrl.toString();
         }
+        entries.append(entry);
```

Comment:
Shouldn't you add only matched entries?

```suggestion
            entry.url = newTrackerUrl.toString();
            entries.append(entry);
        }
```

Reply:
Probably not, the old code does not alter the size of `trackers`. It put all trackers to `torrent->replaceTrackers()`.

Reply:
> Probably not, the old code does not alter the size of `trackers`. It put all trackers back to `torrent->replaceTrackers()`.

Nevermind. I misunderstood the rest of the logic.